### PR TITLE
feat(cast): add `--rpc-timeout` option

### PR DESCRIPTION
**Pull request by PanGan21**
## Motivation
Context: #8959 

## Solution

The following pr adds `--rpc-timeout` option to cast command overriding the default rpc timeout.

### CODE REVIEW

File: crates/cast/bin/cmd/rpc.rs [hunk 1]
```diff
@@ -29,16 +31,29 @@ pub struct RpcArgs {
     #[arg(long, short = 'w')]
     raw: bool,
 
+    /// Timeout for the rpc request
+    ///
+    /// The timeout will be used to override the default timeout for RPC.
+    /// Default value is 45s
+    #[arg(long, env = "ETH_RPC_TIMEOUT")]
+    pub timeout: Option<u64>,
```
Comment (DaniPopes) [inline crates/cast/bin/cmd/rpc.rs:39]: I think we should add this to Config and RpcOpts, so that it's always available in get_provider across the codebase

Comment (DaniPopes) [reply crates/cast/bin/cmd/rpc.rs:39]: cc @mattsse

Comment (PanGan21) [reply crates/cast/bin/cmd/rpc.rs:39]: I tried this as first option in previous commit and it was conflicting with --timeout for other commands.

Comment (PanGan21) [reply crates/cast/bin/cmd/rpc.rs:39]: So it would be possible but the flag should be named differently

Comment (DaniPopes) [reply crates/cast/bin/cmd/rpc.rs:39]: The other commands should also use the config's and RpcOpts timeout

Comment (PanGan21) [reply crates/cast/bin/cmd/rpc.rs:39]: Hi @DaniPopes  just following up on this so we can complete the feature.

Do you believe we should rename the flags of the other commands to something else which will allow to add this timeout to RpcOpts? Their functionality seem different to me so it doesn't seem wise to use the same `RpcOpts::timeout` for everything.

Comment (DaniPopes) [reply crates/cast/bin/cmd/rpc.rs:39]: Yes I think so, a global timeout flag/config makes sense
cc @mattsse @klkvr for more opinions

Comment (mattsse) [reply crates/cast/bin/cmd/rpc.rs:39]: yeah this should indeed be available on RpcOpts level/global level so we can reuse this everywhere

Comment (PanGan21) [reply crates/cast/bin/cmd/rpc.rs:39]: Thanks for the feedback. What about the other `--timeout` flags that conflict with this one if I move it to RpcOpts?

Comment (PanGan21) [reply crates/cast/bin/cmd/rpc.rs:39]: Would it be acceptable if I name it `--request_timeout`?

Comment (grandizzy) [reply crates/cast/bin/cmd/rpc.rs:39]: let's use `--rpc-timeout` for this new one you add and keep `--timeout` for the other for backward compatibility reasons

Comment (PanGan21) [reply crates/cast/bin/cmd/rpc.rs]: Refactored!

**Review by grandizzy — COMMENTED** [review]
left couple of comments, would be great to have confirmation for them before changing

File: crates/cli/src/utils/mod.rs [hunk 2]
```diff
@@ -103,6 +103,11 @@ pub fn get_provider_builder(config: &Config) -> Result<ProviderBuilder> {
         builder = builder.jwt(jwt.as_ref());
     }
 
+    let rpc_timeout = config.get_rpc_timeout()?;
```
[the remark this answers is not in the document]
Comment (PanGan21) [reply crates/cli/src/utils/mod.rs:106]: Don't we want support from Cow when borrowing?

File: crates/cli/src/opts/ethereum.rs [hunk 3]
```diff
@@ -76,6 +83,16 @@ impl RpcOpts {
         Ok(jwt)
     }
 
+    /// Returns the rpc timeout.
```
[the remark this answers is not in the document]
Comment (mattsse) [reply crates/cli/src/opts/ethereum.rs:86]: yeah this should be enough actually

[the remark this answers is not in the document]
Comment (PanGan21) [reply crates/cli/src/opts/ethereum.rs]: Fixed

File: crates/cli/src/utils/mod.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (PanGan21) [reply crates/cli/src/utils/mod.rs]: Fixed!

**Review by grandizzy — APPROVED** [review]
lgtm, thank you! @zerosnacks @yash-atreya could you please check it too before merging?

File: crates/cli/src/opts/ethereum.rs [hunk 4]
```diff
@@ -40,6 +40,13 @@ pub struct RpcOpts {
     /// "0x6bb38c26db65749ab6e472080a3d20a2f35776494e72016d1e339593f21c59bc"]'
     #[arg(long, env = "ETH_RPC_JWT_SECRET")]
     pub jwt_secret: Option<String>,
+
+    /// Timeout for the rpc request
+    ///
+    /// The rpc-timeout will be used to override the default timeout for RPC.
+    /// Default value is 45s
```
Comment (zerosnacks) [inline crates/cli/src/opts/ethereum.rs:47]:
```suggestion
    /// Timeout for the RPC request.
    ///
    /// The` rpc-timeout` will be used to override the default timeout for RPC.
    ///
    /// Default value is 45s.
```

File: crates/config/src/lib.rs [hunk 5]
```diff
@@ -238,6 +238,8 @@ pub struct Config {
     pub eth_rpc_url: Option<String>,
     /// JWT secret that should be used for any rpc calls
     pub eth_rpc_jwt: Option<String>,
+    /// timeout that should be used for any rpc calls
+    pub eth_rpc_timeout: Option<u64>,
```
Comment (zerosnacks) [inline crates/config/src/lib.rs:242]:
```suggestion
    /// Timeout that should be used for any rpc calls
    pub eth_rpc_timeout: Option<u64>,
```

**Review by zerosnacks — APPROVED** [review]
Thanks @PanGan21 for your PR! Applied some minor documentation fixes. LGTM!

**Review by DaniPopes — APPROVED** [review]

**Pull request merged**